src/models/auth.rs

```diff
@@ -49,9 +92,9 @@ pub struct DiscoveryInfo {
 }
 
 #[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
-pub struct LoginResponse {
-    pub user_id: UserId,
+pub struct LoginResponse<'a, 'b> {
+    pub user_id: Cow<'a, UserId>,
     pub access_token: String,
-    pub device_id: DeviceId,
+    pub device_id: Cow<'b, DeviceId>,
```

Comment:
I've spent a bit trying to figure out the Cow here and can't. If it is just a response from the login endpoint, then it doesn't need to be Deserialize and there is no reason to ever own device_id. A Cow just adds overhead. Am I missing something?